driver/controller.go

```diff
@@ -119,6 +155,11 @@ func (s controllerService) CreateVolume(ctx context.Context, req *csi.CreateVolu
 		if node == "" {
 			return nil, status.Errorf(codes.InvalidArgument, "cannot find key '%s' in accessibility_requirements", topolvm.TopologyNodeKey)
 		}
+		if source != nil {
+			if node != sourceVol.Spec.NodeName {
```

Comment:
This logic should be updated according to the following post.
https://github.com/topolvm/topolvm/pull/463#discussion_r895257257

Reply:
@llamerada-jp , the value of node is now set to the value found in the requirements.Preferred or requirements.Requisite. Can there be multiple values for topolvm.TopologyNodeKey?
Should the code be updated to something like this?

```
		for _, topo := range requirements.Preferred {
			if v, ok := topo.GetSegments()[topolvm.TopologyNodeKey]; ok {
				if sourceVol.Spec.NodeName == v {
                                    node = v
				    break
                               }
			}
		}
```
And similarly for the requisites check?

Reply:
I think the current code will work for this scenario where there will only be one entry . Can we keep this as it is?

Reply:
It looks no problem.